cumulusci/tasks/push/push_api.py

```diff
@@ -78,11 +78,10 @@ def version_number(self):
     def get_newer_released_version_objs(self):
         where = f"MetadataPackageId = '{self.package.sf_id}' AND ReleaseState = 'Released' AND "
         version_info = {"major": self.major, "minor": self.minor, "patch": self.patch}
-        where += f"(MajorVersion >= {version_info['major']} OR (MajorVersion = {version_info['major']} AND MinorVersion >= {version_info['minor']}))"
+        where += f"(MajorVersion > {version_info['major']} OR (MajorVersion = {version_info['major']} AND MinorVersion >= {version_info['minor']}))"
```

Comment:
I think we also need 
```suggestion
        where += f"(MajorVersion > {version_info['major']} OR (MajorVersion = {version_info['major']} AND MinorVersion > {version_info['minor']}))"
```

otherwise we'll have a similar problem if we want to push upgrade to a patch version.

Btw, it's partly my fault the logic here got messed up. Sorry about that, I was thinking about it wrong.